### DISCUSSION TITLE
How can I set the node version yarn install is executed with?

### DISCUSSION THREAD

**Discussion by amuttsch** [renovatebot/renovate#8246] [Request Help] [answered]
Hi,

I have a `package.json` in a sub directory which dependencies require node 12, otherwise the dependencies cannot be compiled. Currently everything is build using node 14. Is there a way to force a specific node version to do the renovation task?

I have specified the `engine` field in the `package.json`, but that has no effect unfortunately.

Thanks!

**Comment by rarkins** [comment, marked as answer]
Which type of Renovate are you running?

**Reply by amuttsch** [reply]
I"m using the `renovate/renovate:24.11.5` docker image inside a GitLab CI pipeline

**Reply by viceice** [reply]
@amuttsch checkout https://gitlab.com/renovate-bot/renovate-runner#renovate-runner

You need to use renovate in docker mode or build your own image with node v12

**Reply by amuttsch** [reply]
@viceice Thanks for the tip, I'll have a look. Is there any documentation about this "docker mode"? I could not find this in the docs or via google.

**Reply by viceice** [reply]
https://docs.renovatebot.com/self-hosted-configuration/#binarysource
https://docs.renovatebot.com/self-hosted-configuration/#basedir

the renovate slim image sets `binarySource=docker` by default and needs access to a running docker instance with shared `baseDir`

**Reply by amuttsch** [reply]
This seems to work :tada: Thanks for your help!
